src/plugin/scene/resource/SceneManager.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+#include <optional>
+#include <string>
+
+#include "Engine.hpp"
+
+#include "AScene.hpp"
+
+#include "SceneManager.hpp"
+
+void ES::Plugin::Scene::Resource::SceneManager::Update(ES::Engine::Registry &registry)
+{
+    if (_nextScene.has_value())

```

Comment:
Proposal to avoid yew branches, you can apply it if you like.

```suggestion
    if (!_nextScene.has_value())
    {
        std::cout
            << "[WARNING] ES::Plugin::Scene::Resource::SceneManager: Unable to load next scene: No next scene provided"
            << std::endl;
        return;
    }

    if (_currentScene.has_value())
    {
        std::cout << "[INFO] ES::Plugin::Scene::Resource::SceneManager: Unloading scene: " << _currentScene.value()
                  << std::endl;
        _unloadScene(registry, _currentScene.value());
    }
    std::cout << "[INFO] ES::Plugin::Scene::Resource::SceneManager: Loading scene: " << _nextScene.value()
              << std::endl;
    _loadScene(registry, _nextScene.value());
    _currentScene = _nextScene;
    _nextScene.reset();
```

Reply:
Done! (For the next time, you can select multiple lines of code for a better suggestion)